src/state.rs

```diff
@@ -71,10 +71,43 @@ impl fmt::Display for Upstream {
     }
 }
 
+// Workload information that a specific proxy instance represents. This is used to cross check
+// with the workload fetched using destination address when making RBAC decisions.
+#[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct WorkloadInfo {
+    pub name: String,
+    pub namespace: String,
+    pub trust_domain: String,
+    pub service_account: String,
+}
+
+impl WorkloadInfo {
+    pub fn new(
+        name: String,
+        namespace: String,
+        trust_domain: String,
+        service_account: String,
+    ) -> Self {
+        Self {
+            name,
+            namespace,
+            trust_domain,
+            service_account,
+        }
+    }
+
+    pub fn matches(&self, w: &Workload) -> bool {
```

Comment:
Deferrable nit - why not just implement `Eq`?

Reply:
different types. i'm comparing this to `Workload` not `WorkloadInfo`

Reply:
derp - yep